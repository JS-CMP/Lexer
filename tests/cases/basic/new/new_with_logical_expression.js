function A() { this.name = "A"; }
function B() { this.name = "B"; }

var useA = false;
console.log(new (useA && A || B)()); // NEW((useA && A || B));
