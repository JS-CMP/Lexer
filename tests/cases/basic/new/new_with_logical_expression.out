#include "types/JsAny.hpp"
#include "global/global.hpp"
#include "types/objects/Types.hpp"
#include "customOperators/CustomOperators.hpp"

int main() {JS::Any A, B, useA;A = JS::Any(JS::Function([&](const JS::Any &thisArg, const JS::Any &arguments) -> JS::Any {thisArg["name"]=JS::Any("A");return JS::Any();}));B = JS::Any(JS::Function([&](const JS::Any &thisArg, const JS::Any &arguments) -> JS::Any {thisArg["name"]=JS::Any("B");return JS::Any();}));useA=JS::Any(false);console["log"](NEW((useA&&A||B)));}
